src/services/schedule.js

```diff
@@ -3,296 +3,332 @@ const _ = require('lodash');
-            // console.log('Branches: ');
-            for (branch of allBranches) {
-                let count = 0;
-                // console.log(branch);
-                for (task of branch) {
-                    if (task.scheduled_time != null) {
-                        count++;
-                    }
-                }
-                maxCount = count > maxCount ? count : maxCount;
-            }
-
-            // Choose solution with max number
-            for (branch of allBranches) {
-                let count = 0;
-                for (task of branch) {
-                    if (task.scheduled_time != null) {
-                        count++;
-                    }
-                }
-
-                if (count == maxCount) {
-                    // console.log('\nTask: ' + currentTask.id);
-                    // console.log('Returning');
-                    // console.log(branch);
-                    return branch;
-                }
-            }
+      }
+
+      if (allBranches.length == 0) {
+        // It's impossible to schedule this task, so let all other tasks try
+        currentTask.scheduled_time = null;
+        let restTasks = tasks.slice(1);
+        return [currentTask].concat(solve(restTasks, freeIntervals));
+      }
+
+      // Go through all branches and find max number of tasks scheduled
+      let maxCount = 0;
+      // console.log('\nTask: ' + currentTask.id);
+      // console.log('Branches: ');
+      for (branch of allBranches) {
+        let count = 0;
+        // console.log(branch);
+        for (task of branch) {
+          if (task.scheduled_time != null) {
+            count++;
+          }
         }
-
-        const freeIntervals = await getFreeIntervals(startTime, localTZ, hoursOfOpStart, hoursOfOpEnd, busyIntervals);
-        const result = solve(tasks, freeIntervals);
-        for (let i = 0; i < result.length; i++) {
-            if (result[i].scheduled_time != null) {
-                result[i].scheduled_time = moment.unix(result[i].scheduled_time).toISOString();
-            }
+        maxCount = count > maxCount ? count : maxCount;
+      }
+
+      // Choose solution with max number
+      for (branch of allBranches) {
+        let count = 0;
+        for (task of branch) {
+          if (task.scheduled_time != null) {
+            count++;
+          }
         }
 
-        return result;
+        if (count == maxCount) {
+          // console.log('\nTask: ' + currentTask.id);
+          // console.log('Returning');
+          // console.log(branch);
+          return branch;
+        }
+      }
     }
 
-    return {
-        getFreeIntervals, // returned only for debugging purposes!
-        scheduleTasks,
+    // Insert a busy interval at the last task due date
+    const lastTask = tasks[tasks.length - 1];
+    const lastTaskDue = lastTask.due_date;
+    const endBusyInterval = [lastTaskDue, lastTaskDue];
+    busyIntervals.push(endBusyInterval);
+
```

Comment:
Lines 291-295.
Insert a busy "interval" of 0 seconds at the due date of the last task to ensure that the free time finder works properly. Resolves issue of being unable to schedule on an empty calendar.